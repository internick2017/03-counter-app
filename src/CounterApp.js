import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value = 5 }) => {

    // const [nombre, setNombre] = useState('Goku');
    // console.log(nombre, setNombre);
    const [counter, setCounter] = useState(value); // retorna un arreglo con 2 valores


    // handleAdd
    const handleAdd = () => {
        setCounter(counter + 1)
        // setCounter((c) => c + 1)
    }
    const handleSubstract = () => {
        setCounter(counter - 1)
    }
    const handleReset = () => {
        setCounter(value)
    }

    return (

        <>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>

            <button onClick={handleAdd} >+1</button>
            <button onClick={handleReset} >Reset</button>
            <button onClick={handleSubstract} >-1</button>
        </>

    );
}

CounterApp.propTypes = {
    value: PropTypes.number
}

export default CounterApp;