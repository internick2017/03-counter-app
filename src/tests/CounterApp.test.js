import React from 'react';
import '@testing-library/jest-dom';
import CounterApp from "../CounterApp";
import { shallow } from 'enzyme';

describe('Pruebas counter App tarea', () => {

    let wrapper = shallow(<CounterApp />);

    beforeEach(() => { // hacer antes de cada prueba
        wrapper = shallow(<CounterApp />)
    })

    test('deve mostrar counter app correctamente y hacer snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('deve mostrat el valor por defecto de 100', () => {

        const value = 100;
        const wrapper = shallow(
            <CounterApp
                value={value}
            />)

        const valueInH2 = wrapper.find('h2').text()
        const valueInt = parseInt(valueInH2);
        expect(valueInt).toBe(value);
    })

    test('deve de incrementar con el boton +1', () => {
        const valueInH2 = wrapper.find('h2').text();
        const valueInt = parseInt(valueInH2);
        const button1 = wrapper.find('button').at(0);
        const simulateButton = button1.simulate('click');
        const newValueInH2 = wrapper.find('h2').text();
        const newValueInt = parseInt(newValueInH2);

        expect(newValueInt).toBe(valueInt + 1)
    })

    test('deve de restar con el boton -1', () => {
        const valueInH2 = wrapper.find('h2').text();
        const valueInt = parseInt(valueInH2);
        const button2 = wrapper.find('button').at(2);
        const simulateButton = button2.simulate('click');
        const newValueInH2 = wrapper.find('h2').text();
        const newValueInt = parseInt(newValueInH2);
        expect(newValueInt).toBe(valueInt - 1)
    })

    test('deve mostar el valor por defecto cuando clica e el boton de reset', () => {
        const value = 105;
        const wrapper = shallow(
            <CounterApp
                value={value}
            />)
        const buttonMore1 = wrapper.find('button').at(0);
        const buttonReset = wrapper.find('button').at(1);
        const simulateButton = buttonMore1.simulate('click');
        const simulateButton2 = buttonMore1.simulate('click');
        const simulateButtonReset = buttonReset.simulate('click');
        const newValueInH2 = wrapper.find('h2').text();
        const newValueInt = parseInt(newValueInH2);
        expect(newValueInt).toBe(value);
    })




})
