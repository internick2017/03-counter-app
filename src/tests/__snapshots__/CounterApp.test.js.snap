// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas counter App tarea deve mostrar counter app correctamente y hacer snapshot 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    5
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
